Tests/StreamChatUITests/Utils/DefaultMarkdownFormatter_Tests.swift

```diff
@@ -0,0 +1,252 @@
+//
+// Copyright © 2022 Stream.io Inc. All rights reserved.
+//
+
+import StreamChatTestHelpers
+@testable import StreamChatUI
+import XCTest
+
+final class DefaultMarkdownFormatter_Tests: XCTestCase {
+    var sut: DefaultMarkdownFormatter!
+
+    override func setUpWithError() throws {
+        try super.setUpWithError()
+        sut = DefaultMarkdownFormatter()
+    }
+
+    override func tearDownWithError() throws {
+        sut = nil
+        try super.tearDownWithError()
+    }
+    
+    func test_containsMarkdown_whenCheckOnAStringWithNoMarkdown_thenReturnsFalse() {
+        // GIVEN
+        let stringWithNoMarkdown = "Hello, This is a test String"
+
+        // WHEN
+        let containsMarkdown = sut.containsMarkdown(text: stringWithNoMarkdown)
+
+        // THEN
+        XCTAssertEqual(false, containsMarkdown)
+    }
+    
+    func test_containsMarkdown_whenCheckForItalicEmphasis_thenReturnsTrue() {
```

Comment:
It does seem kind of verbose, but since they are unit tests, is ideal to test just one specific behavior. These are to make sure that we detect each of the markdown symbols properly. 
With the comments you refer to the // Given, When, Then? This is the approach we are following with the team for unit tests.
Since we have the validation, empty text doesn’t even get to be formatted with the MarkdownFormatter nor we have invalid markdown. Basically with the regex validation, incomplete or invalid Regex syntax is just set to the textView as it is.